src/OpalCompiler-Core/OCASTSemanticAnalyzer.class.st

```diff
@@ -214,15 +214,18 @@ OCASTSemanticAnalyzer >> visitMethodNode: aMethodNode [
 
 { #category : #visiting }
 OCASTSemanticAnalyzer >> visitPragmaNode: aPragmaNode [
+
+	| varNode |
 	super visitPragmaNode: aPragmaNode.
+	aPragmaNode selector = #compilerOptions: ifTrue: [ 
+		aPragmaNode asPragma sendTo:
+			aPragmaNode methodNode compilationContext ].
```

Comment:
Maybe it makes sense to push this knowledge about such special pragmas into superclass to have #visitCompilerOptionsPragma: and #visitPrimitiveErrorPragma: